File: sample/fullpath.rb
#! /usr/local/bin/ruby
# convert ls-lR filename into fullpath.

if ARGV[0] =~ /-p/
  ARGV.shift
  path = ARGV.shift 
end

if path == nil
  path = ""
elsif path !~ /\/$/
  path += "/"
end

while gets()
  if /:$/
    path = $_.chop.chop + "/"
  elsif /^total/ || /^d/
  elsif /^(.*\d )(.+)$/
    print($1, path, $2, "\n")
  end
end
    

File: sample/fullpath.pl
#! /usr/local/bin/perl
# convert ls-lR filename into fullpath.

$path = shift;
if (!defined $path) {
    $path = "";
}
elsif ($path !~ /\/$/) {
  $path .= "/"
}

while (<>) {
    if (/:$/) {
	chop; chop;
	$path = $_ . "/";
    } elsif (/^total/ || /^d/) {
	next;
    } elsif (/^(.*\d )(.+)$/) {
	print $1, $path, $2, "\n";
    }
}
    
